### PR TITLE
1643 annotations with measures

### PR DESCRIPTION
**Pull request by ramtob**

### CODE REVIEW

File: src/app/@ansyn/core/models/visualizers/annotations.model.ts [hunk 1]
```diff
@@ -7,10 +7,16 @@ export interface IAnnotationsContextMenuBoundingRect {
 }
 
 export interface IAnnotationsContextMenuEvent {
```
Comment (yairtawil) [inline src/app/@ansyn/core/models/visualizers/annotations.model.ts]: rename

File: src/app/@ansyn/plugins/openlayers/visualizers/tools/annotations.visualizer.ts [hunk 2]
```diff
@@ -247,12 +266,13 @@ export class AnnotationsVisualizer extends EntitiesVisualizer {
 		}
 		const selectedFeature = AnnotationsVisualizer.findFeatureWithMinimumArea(data.selected);
 		const boundingRect = this.getFeatureBoundingRect(selectedFeature);
-		const { id } = selectedFeature.getProperties();
+		const { id, showMeasures } = selectedFeature.getProperties();
```
Comment (yairtawil) [inline src/app/@ansyn/plugins/openlayers/visualizers/tools/annotations.visualizer.ts]: use `this.getEntity(selectedFeature)`

File: src/app/@ansyn/plugins/openlayers/visualizers/tools/annotations.visualizer.ts [hunk 3]
```diff
@@ -411,6 +435,109 @@ export class AnnotationsVisualizer extends EntitiesVisualizer {
...
+			return style;
+		}
+	}
+
+	getMeasuresAsStyles(feature: OlFeature): OlStyle[] {
+		const { mode } = feature.getProperties();
+		const view = (<any>this.iMap.mapObject).getView();
+		const projection = view.getProjection();
+		let coordinates: any[] = [];
+		let moreStyles: OlStyle[] = [];
```
Comment (yairtawil) [inline src/app/@ansyn/plugins/openlayers/visualizers/tools/annotations.visualizer.ts]: const

File: src/app/@ansyn/map-facade/components/annotation-context-menu/annotation-context-menu.component.html [hunk 4]
```diff
@@ -2,6 +2,8 @@
 	[ngStyle]="contextMenuWrapperStyle">
 
 	<div class="remove-feature-wrapper" *ngIf="!fromHover">
+		<!-- Don't change "mousedown" to "click"! -->
+		<button><i class="icon-tools-ruler" (mousedown)="toggleMeasures($event)"></i></button>
```
Comment (yairtawil) [inline src/app/@ansyn/map-facade/components/annotation-context-menu/annotation-context-menu.component.html]: measure 'active' 'inactive'

File: src/app/@ansyn/core/models/visualizers/annotations.model.ts (hunk 1 above)
Comment (ramtob) [reply src/app/@ansyn/core/models/visualizers/annotations.model.ts]: done

File: src/app/@ansyn/map-facade/components/annotation-context-menu/annotation-context-menu.component.html (hunk 4 above)
Comment (ramtob) [reply src/app/@ansyn/map-facade/components/annotation-context-menu/annotation-context-menu.component.html]: done

File: src/app/@ansyn/plugins/openlayers/visualizers/tools/annotations.visualizer.ts (hunk 2 above)
Comment (ramtob) [reply src/app/@ansyn/plugins/openlayers/visualizers/tools/annotations.visualizer.ts]: done

File: src/app/@ansyn/ansyn/app-effects/effects/layers.app.effects.ts [hunk 5]
```diff
@@ -55,9 +59,15 @@ export class LayersAppEffects {
 			const layer = layers
 				.filter(({ type }) => type === LayerType.annotation)
 				.find((layer: ILayer) => layer.data.features.some(({ properties }: Feature<any>) => properties.id === action.payload.featureId));
-			const feature = layer.data.features.find(({ properties }: Feature<any>) => properties.id === action.payload.featureId);
-			feature.properties = {...feature.properties, ...action.payload.properties};
-			return new UpdateLayer(layer);
+			return new UpdateLayer({
+				...layer,
+				data: {
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/app-effects/effects/layers.app.effects.ts]: HOPAAAAA

**Pull request merged**